src/extension.ts

```diff
@@ -1,39 +1,79 @@
 import * as vscode from 'vscode';
```

Comment:
use ExtractorAbstract for interchangability, dont just define your own listeners!